### PR TITLE
MEN-4580: Install demo cert in local trust store

### PR DESCRIPTION
**Pull request by lluiscampos**
Changelog: mender setup: when configuring for demo using self-signed
certificate, install the certificate in the local trust store so that
all components in the system (namely, Mender addons) can trust the
Mender server without extra configuration.

### CODE REVIEW

**Comment by lluiscampos** [comment]
Missing unit tests.

**Comment by lluiscampos** [comment]
> @lluiscampos, Let me know if you want to start the integration pipeline by mentioning me and the command "start pipeline".

@mender-test-bot Please do that.

File: cli/commands.go [hunk 1]
```diff
@@ -258,6 +261,46 @@ func initDaemon(config *conf.MenderConfig,
 	return daemon, nil
 }
 
+func checkDemoCert() {
+	trustStoreCert, err := os.Open(getLocalTrustMenderCertPath())
```
Comment (tranchitella) [inline cli/commands.go:265]: You are calling `getLocalTrustMenderCertPath` several times; shouldn't we create a variable and reuse it?

File: cli/commands.go [hunk 2]
```diff
@@ -258,6 +261,46 @@ func initDaemon(config *conf.MenderConfig,
...
+	if err != nil {
+		log.Debugf("Could not open Mender certificate in local trust store: %s", err.Error())
+		return
+	}
+	defer trustStoreCert.Close()
+
+	demoCert, err := os.Open(getMenderDemoCertPath())
```
Comment (tranchitella) [inline cli/commands.go:272]: You are calling `getMenderDemoCertPath` several times; shouldn't we create a variable and reuse it?

File: cli/setup.go [hunk 3]
```diff
@@ -65,6 +68,20 @@ const ( // state enum
...
+	DefaultMenderDemoCertDir   = "/usr/share/doc/mender-client/examples"
+	DefaultLocalTrustMenderDir = "/usr/local/share/ca-certificates/mender"
+)
+
+func getMenderDemoCertPath() string {
+	return path.Join(DefaultMenderDemoCertDir, "demo.crt")
+}
+
+func getLocalTrustMenderCertPath() string {
+	return path.Join(DefaultLocalTrustMenderDir, "server.cert")
```
Comment (tranchitella) [inline cli/setup.go:82]: server.cert or server.crt?

File: cli/setup.go [hunk 4]
```diff
@@ -885,3 +909,52 @@ func (opts *setupOptionsType) maybeAddHostLookup() {
...
+	_, err = os.Stat(dir)
+	if os.IsNotExist(err) {
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			return errors.Wrapf(err,
+				"Cannot create directory %q", dir)
+		}
+	}
+
+	d, err := os.OpenFile(getLocalTrustMenderCertPath(), os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0755)
```
Comment (tranchitella) [inline cli/setup.go:937]: Why 0755 mode?

File: cli/setup.go (hunk 3 above)
Comment (lluiscampos) [reply cli/setup.go:82]: `.crt`, thanks!

File: cli/setup.go (hunk 4 above)
Comment (lluiscampos) [reply cli/setup.go:937]: Changing it to `444` to align it with yocto install. Thanks.

**Comment by lluiscampos** [comment]
> @lluiscampos, Let me know if you want to start the integration pipeline by mentioning me and the command "start pipeline".

@mender-test-bot Just in case, run them again...

**Comment by lluiscampos** [comment]
I have reworked the minor things that @tranchitella pointed out, see [diff here](https://github.com/mendersoftware/mender/compare/24f0ff6ac05edbadc5ac900c3c7a1ae62a8caa0c..ef71d96fe8d96adf4a70f3198f65154f1cdb7fc7). The previous build (see [pipeline here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/278874305)) passed.

@kacf If this turns out to be the only blocker for the release, I support merging it without completing the current build - IMO the rework changes are very low risk.

**Review by tranchitella — APPROVED** [review]
Let's merge it ASAP

**Comment by kacf** [comment]
I can cherry-pick this.

**Comment by kacf** [comment]
Oh, it's done, never mind.

**Pull request merged**